db/queries.py

```diff
@@ -3,44 +3,71 @@
 
 
 def get_messages_that_contain_word(word: str) -> list[Message]:
-    pass
+    return list(Message.objects.filter(
+        text__icontains=word
+    ))
 
 
 def get_untitled_chats() -> list[Chat]:
-    pass
+    return Chat.objects.filter(
+        title__startswith="Untitled"
+    )
 
 
 def get_users_who_sent_messages_in_2015() -> list[str]:
-    pass
+    return list(User.objects.filter(
+        message__sent__istartswith="2015"
+    ).values_list("first_name", "last_name"))
 
 
 def get_actual_chats() -> list[Chat]:
-    pass
+    return Chat.objects.filter(
+        message__sent__gte="2020-01-01"
+    ).distinct()
 
 
 def get_messages_contain_authors_first_name() -> list[Message]:
-    pass
+    return Message.objects.filter(
+        text__contains=F("user__first_name")
+    )
 
 
 def get_users_who_sent_messages_starts_with_m_or_a() -> list[User]:
-    pass
+    return User.objects.filter(
+        Q(message__text__istartswith="a") | Q(message__text__istartswith="m")
+    )
 
 
 def get_delivered_or_admin_messages() -> list[Message]:
-    pass
+    return list(Message.objects.filter(
+        Q(user__username__startswith="admin") | Q(is_delivered=True)
+    ))
 
 
 def get_count_messages_sent_by_first_name(first_name: str) -> int:
-    pass
+    return Message.objects.filter(
+        user__first_name=first_name
+    ).count()
 
 
 def get_top_users_by_number_of_the_messages() -> list[User]:
-    pass
+    return list(User.objects.annotate(
+        num_messages=Count("message")
+    ).order_by("-num_messages")[:3])
 
 
 def get_last_5_messages_dicts() -> list[dict]:
-    pass
+    return [{"from": message.user.username, "text": message.text}
+            for message in Message.objects.select_related(
+            "user"
+            ).order_by("-sent")[:5]]
```

Comment:
В тебе нижче краще розкадно все по рядкам, так і тут зроби
![image](https://user-images.githubusercontent.com/31931886/212895598-5afb40f4-278b-47e7-9f9a-0edef93adfd6.png)
```suggestion
    return [
        {"from": message.user.username, "text": message.text}
         for message in (
            Message.objects
            .select_related("user")
            .order_by("-sent")[:5]
         )
         ]
```

Reply:
якось так